File: extract_primary_html_content.py
# This code assumes you're in n8n, modify the variables according to your usecase


import re

def extract_primary_text(html):
    # Remove scripts, styles, comments, head, and other non-content
    html = re.sub(r'<script[^>]*>[\s\S]*?</script>', '', html, flags=re.IGNORECASE | re.DOTALL)
    html = re.sub(r'<style[^>]*>[\s\S]*?</style>', '', html, flags=re.IGNORECASE | re.DOTALL)
    html = re.sub(r'<head[^>]*>[\s\S]*?</head>', '', html, flags=re.IGNORECASE | re.DOTALL)
    html = re.sub(r'<!--[\s\S]*?-->', '', html, flags=re.DOTALL)
    html = re.sub(r'<noscript[^>]*>[\s\S]*?</noscript>', '', html, flags=re.IGNORECASE | re.DOTALL)
    html = re.sub(r'<iframe[^>]*>[\s\S]*?</iframe>', '', html, flags=re.IGNORECASE | re.DOTALL)
    html = re.sub(r'<object[^>]*>[\s\S]*?</object>', '', html, flags=re.IGNORECASE | re.DOTALL)
    html = re.sub(r'<embed[^>]*>[\s\S]*?</embed>', '', html, flags=re.IGNORECASE | re.DOTALL)

    # Remove remaining tags, replacing with space to preserve separation
    html = re.sub(r'<[^>]+>', ' ', html)

    # Normalize whitespace
    return re.sub(r'\s+', ' ', html).strip()

# In n8n Python Code node, input is 'items' (list of dicts)
# Assuming HTML is in item['json']['data'] (adjust field name if needed)
return [{'json': {'text': extract_primary_text(item['json']['data'])}} for item in items]

File: extract_primary_html_content.js
// This code assumes you're in n8n, modify the variables according to your usecase

function extractPrimaryText(html) {
  // Remove scripts, styles, comments, head, and other non-content
  html = html.replace(/<script[^>]*>[\s\S]*?<\/script>/gi, '')
             .replace(/<style[^>]*>[\s\S]*?<\/style>/gi, '')
             .replace(/<head[^>]*>[\s\S]*?<\/head>/gi, '')
             .replace(/<!--[\s\S]*?-->/g, '')
             .replace(/<noscript[^>]*>[\s\S]*?<\/noscript>/gi, '')
             .replace(/<iframe[^>]*>[\s\S]*?<\/iframe>/gi, '')
             .replace(/<object[^>]*>[\s\S]*?<\/object>/gi, '')
             .replace(/<embed[^>]*>[\s\S]*?<\/embed>/gi, '');

  // Remove remaining tags, replacing with space to preserve separation
  html = html.replace(/<[^>]+>/g, ' ');

  // Normalize whitespace
  return html.trim().replace(/\s+/g, ' ');
}

return $input.all().map(item => ({ json: { text: extractPrimaryText(item.json.data) } }));
